hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/contracts/operation/HederaOperationUtilV038.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (C) 2021-2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.service.mono.contracts.operation;
+
+import static com.hederahashgraph.api.proto.java.HederaFunctionality.ContractCall;
+
+import com.hedera.node.app.service.evm.contracts.operations.HederaExceptionalHaltReason;
+import com.hedera.node.app.service.mono.contracts.sources.EvmSigsVerifier;
+import com.hedera.node.app.service.mono.state.merkle.MerkleAccount;
+import com.hedera.node.app.service.mono.store.contracts.HederaStackedWorldStateUpdater;
+import com.hedera.node.app.service.mono.store.contracts.HederaWorldState;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.TreeMap;
+import java.util.function.BiPredicate;
+import java.util.function.BooleanSupplier;
+import java.util.function.LongSupplier;
+import java.util.function.Predicate;
+import java.util.function.Supplier;
+import org.apache.commons.lang3.tuple.MutablePair;
+import org.apache.tuweni.bytes.Bytes32;
+import org.apache.tuweni.units.bigints.UInt256;
+import org.hyperledger.besu.datatypes.Address;
+import org.hyperledger.besu.evm.frame.ExceptionalHaltReason;
+import org.hyperledger.besu.evm.frame.MessageFrame;
+import org.hyperledger.besu.evm.operation.Operation;
+
+/** Utility methods used by Hedera adapted {@link Operation} */
+public final class HederaOperationUtilV038 {
```

Comment:
I fully agree with everything you said. 
I intend to go over each new version of a utility/opcode/processor class and:
1. describe what's changed in this new version and why
2. think about cleaning up the code and optimisations such as you described
This can probably be done in a future refactoring PR if we want to push these changes out today.

Reply:
This is fine with me!  Make sure there's an issue then you can resolve this conversation.  Thanks!